lib/cn/cn.c

```diff
@@ -1057,6 +1023,49 @@ cn_open(
     rp->cn_maint_disable = rp->cn_maint_disable || cn->cn_replay ||
         rp->cn_diag_mode || rp->read_only;
 
+    /* Roll up node stats to get KVS stats for the open kvs log message.
+     */
+    {
+        const char *suffixes = "bkmgtp";
+        struct kvset_stats kvs_stats = { 0 };
+        struct cn_node_stats ns;
+        ulong hshift, kshift, vshift;
+        char hszsuf, kszsuf, vszsuf;
+
+        list_for_each_entry(tn, &cn->cn_tree->ct_nodes, tn_link) {
+            cn_node_stats_get(tn, &ns);
+            kvset_stats_add(&ns.ns_kst, &kvs_stats);
+        }
+
+        hshift = ilog2(kvs_stats.kst_halen | 1) / 10;
+        kshift = ilog2(kvs_stats.kst_kalen | 1) / 10;
+        vshift = ilog2(kvs_stats.kst_valen | 1) / 10;
+
+        hszsuf = suffixes[hshift];
+        kszsuf = suffixes[kshift];
+        vszsuf = suffixes[vshift];
+
+        log_info(
+            "opened kvs %s/%s cnid %lu pfx_len %u vcomp %u,%u"
+            " hb %lu%c/%lu kb %lu%c/%lu vb %lu%c/%lu %s%s%s%s%s%s",
+            cn->cn_kvdb_alias, cn->cn_kvsname, (ulong)cnid,
+            cn->cp->pfx_len,
+            cn->rp->value_compression, cn->rp->vcompmin,
+            (ulong)kvs_stats.kst_halen >> (hshift * 10), hszsuf, (ulong)kvs_stats.kst_hblks,
+            (ulong)kvs_stats.kst_kalen >> (kshift * 10), kszsuf, (ulong)kvs_stats.kst_kblks,
+            (ulong)kvs_stats.kst_valen >> (vshift * 10), vszsuf, (ulong)kvs_stats.kst_vblks,
+            rp->mclass_policy,
+            rp->cn_maint_disable ? " !maint" : "",
+            rp->cn_diag_mode ? " diag" : "",
+            rp->read_only ? " rdonly" : "",
```

Comment:
Just spell out read only

Reply:
I didn't edit the log message.  That terminology comes from flags used with `open(2)`.